QWeb/internal/browser/firefox.py

```diff
@@ -71,10 +71,9 @@ def open_browser(profile_dir: Optional[str] = None,
     options.set_preference("privacy.socialtracking.block_cookies.enabled", False)
     kwargs = {k.lower(): v for k, v in kwargs.items()}  # Kwargs keys to lowercase
     if 'prefs' in kwargs:
-        if isinstance(kwargs.get('prefs'), dict):
-            prefs = kwargs.get('prefs')
-        else:
-            prefs = util.prefs_to_dict(str(kwargs.get('prefs')).strip())
+        tmp_prefs = kwargs.get('prefs')
+        prefs = util.validate_prefs(tmp_prefs)
```

Comment:
Check review comment (change suggestion) to util.validate_prefs()

With that change this can be simplified to just
prefs = util.prefs_to_dict(tmp_prefs)

Additionally pref checks should be harmonized to use either `tmp_prefs` or directly `kwargs.get('prefs')`

Reply:
not changed in this pr, so will make a specific refactor task about this

Reply:
changed validate_prefs to parse_prefs